app/assets/javascripts/controllers/ems_common/ems_common_form_controller.js

```diff
@@ -336,16 +358,24 @@ ManageIQ.angular.app.controller('emsCommonFormController', ['$http', '$scope', '
     }
   };
 
-  $scope.metricSelectionChanged = function() {
-    if ($scope.emsCommonModel.metrics_selection == "disabled") {
+  $scope.tabSelectionChanged = function(tabSelector, selection) {
+    if (selection == "disabled") {
       $scope.changeAuthTab('default');
       angular.element('.nav-tabs a[href="#default"]').tab('show');
-      angular.element("#metrics_tab").hide();
+      angular.element(tabSelector).hide();
     } else {
-      angular.element("#metrics_tab").show();
+      angular.element(tabSelector).show();
     }
```

Comment:
This code is OK for the first pass, since it has already been tested.
But I'm wondering if we should move this code in a directive later.

Reply:
Yes 👍 , the whole ruby `= select_tag(...` in the .haml file, should be in angular so will not need this hack :-(